sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -154,6 +155,22 @@ class CacheManager extends Logging {
     }
   }
 
+  // Analyzes column statistics in the given cache data
+  private[sql] def analyzeColumnCacheQuery(
+      sparkSession: SparkSession,
+      cachedData: CachedData,
+      column: Seq[Attribute]): Unit = {
+    val relation = cachedData.cachedRepresentation
+    val (rowCount, newColStats) =
+      CommandUtils.computeColumnStats(sparkSession, relation, column)
+    val oldStats = relation.statsOfPlanToCache
+    val newStats = oldStats.copy(
+      rowCount = Some(rowCount),
+      attributeStats = AttributeMap((oldStats.attributeStats ++ newColStats).toSeq)
+    )
+    relation.statsOfPlanToCache = newStats
```

Comment:
@dongjoon-hyun On second thought, we need lock here as you suggested before. Could you check again? https://github.com/apache/spark/pull/24178

Reply:
Yes. `Locking` is more conservative and the one I preferred. BTW, what's the change in your mind?